CNV-27495: new virtctl manifest commands (#63770)
openshift/openshift-docs: modules/virt-virtctl-commands.adoc

[id='vm-volume-export-commands_{context}']
== VM export commands

> Use `virtctl vmexport` commands to create, download, or delete a volume exported from a VM, VM snapshot, or persistent volume claim (PVC). Certain manifests may also contain a header secret, which grants access to the endpoint in order to import a disk image in a format that {VirtProductName} can use.

Reviewer: ```suggestion
Use `virtctl vmexport` commands to create, download, or delete a volume exported from a VM, VM snapshot, or persistent volume claim (PVC). Certain manifests might also contain a header secret, which grants access to the endpoint to import a disk image in a format that {VirtProductName} can use.
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#i

|`virtctl vmexport download <vmexport_name> --<vm\|snapshot\|pvc>=<object_name> --output=<output_file> --volume=<volume_name>`
|Create a `VirtualMachineExport` CR and then download the volume defined in the CR.

|`virtctl vmexport download export --manifest`

> |Retrieve the manifest for an existing export. The manifest does not include the header secret.

Reviewer 2: Do we want to explain what a 'header secret' is? It is a token that allows access to the endpoint in a format that can used by OCP-V to import the disk image.

Reviewer 3: What about the `service-url` flag? Maybe we could add some examples for it too.

Reviewer 2: I guess we could, but in OpenShift that should not be needed since it is automatically detected from the Route

Author: @awels 
Am I understanding correctly: 
Yes to adding a description of header secret?
No to adding service-flag info?

Reviewer 2: Yes on adding the description of the header secret since that is not a standard thing, so people won't be familiar with it. I am on the fence on the service-url flag, in general people shouldn't need to use it because it is auto detected, but I can imagine scenarios where it would be needed, so I am not entirely sure.

|`virtctl vmexport download export --manifest --vm=example`
|Create a VM export for a VM example, and retrieve the manifest. The manifest does not include the header secret.

|`virtctl vmexport download export --manifest --snap=example`
|Create a VM export for a VM snapshot example, and retrieve the manifest. The manifest does not include the header secret.

|`virtctl vmexport download export --manifest --include-secret`
|Retrieve the manifest for an existing export. The manifest includes the header secret.

|`virtctl vmexport download export --manifest --manifest-output-format=json`
|Retrieve the manifest for an existing export in json format. The manifest does not include the header secret.

|`virtctl vmexport download export --manifest --include-secret --output=manifest.yaml`

> |Retrieve the manifest for an existing export. The manifest will include the header secret and write it to the file specified.

Reviewer: ```suggestion
|Retrieve the manifest for an existing export. The manifest includes the header secret and writes it to the file specified.
```
If the manifest will include the header secret some time in the future, then ignore this suggestion. However,, the suggestion uses the active voice like the instance on line 188.
